lib/engine/action/place_token.rb

```diff
@@ -5,22 +5,26 @@
 module Engine
   module Action
     class PlaceToken < Base
-      attr_reader :city, :slot
+      attr_reader :city, :slot, :token
 
-      def initialize(entity, city, slot)
+      def initialize(entity, city, slot, token_id = nil)
         @entity = entity
         @city = city
         @slot = slot
+        # After processing the action the token may no longer be in the entity
+        @token_id = token_id
+        @token = @entity.tokens[token_id] unless token_id.nil?
```

Comment:
if token_id

but as you said this may not exist, so do we need to keep track in the game like the other objects?

token_by_id

Reply:
I was thinking about this, I'd like to add a token type (:normal and :neutral), and then instead of passing in the index, pass in the type to lay. I think it'd make it cleaner.
I'd also then group tokens by type rather than corp.

If I did it this way could remove the need for 1882 operating class by making that code specific to type :neutral in the normal operating class.

Is this a sensible approach?

Reply:
Also the neutral token not blocking would be easier to code

Reply:
we can give that a shot

Reply:
seems to work fine